soc/usbexecd.c

```diff
@@ -446,7 +447,8 @@ void pipebuf_in_ack(pipebuf_t* pb, size_t ack_number_size) {
     // If the socket was closed prematurely
     if (sock_closed == -1) {
         // Return back to the event loop
-        fatal("Remote socket closed in the middle of sending ACK length bytes");
+        debug("Remote socket closed in the middle of sending ACK length bytes");
+        return handle_closed_spid_socket();
```

Comment:
Instead of totally killing the process, this change will just clear all current state (processes, streams, etc.) and wait for a new socket.